tests/integration/lifecycle/test_lifecycle_partitions.py

```diff
@@ -0,0 +1,390 @@
+        parts_yaml = textwrap.dedent(
+            """
+            parts:
+              foo:
+                plugin: dump
+                source: foo
+              bar:
+                plugin: dump
+                source: bar
+            """
+        )
+        Path("foo").mkdir()
+        Path("foo/foo.txt").touch()
+        Path("bar").mkdir()
+        Path("bar/bar.txt").touch()
+
+        parts = yaml.safe_load(parts_yaml)
+
+        self._lifecycle = craft_parts.LifecycleManager(
+            parts,
+            application_name="test_clean",
+            cache_dir=new_dir,
+            partitions=["default", "other"],
+        )
+
+        # pylint: enable=attribute-defined-outside-init
+
+    @pytest.mark.parametrize(
+        "step,test_dir,state_file",
+        [
+            (Step.PULL, "parts/foo/src", "pull"),
+            (Step.BUILD, "parts/foo/install", "build"),
+            (Step.STAGE, "stage", "stage"),
+            (Step.PRIME, "prime", "prime"),
+        ],
+    )
+    def test_clean_step(self, step, test_dir, state_file):
+        actions = self._lifecycle.plan(step)
+
+        with self._lifecycle.action_executor() as ctx:
+            ctx.execute(actions)
+
+        assert Path(test_dir, "foo.txt").is_file()
+        assert Path("parts/foo/state", state_file).is_file()
+
+        self._lifecycle.clean(step, part_names=["foo"])
+
+        assert Path(test_dir, "foo.txt").is_file() is False
+        assert Path("parts/foo/state", state_file).is_file() is False
+
+    def test_clean_default(self):
+        actions = self._lifecycle.plan(Step.PRIME)
+
+        with self._lifecycle.action_executor() as ctx:
+            ctx.execute(actions)
+
+        assert Path("parts/foo/src/foo.txt").is_file()
+        assert Path("parts/foo/install/foo.txt").is_file()
+        assert Path("stage/foo.txt").is_file()
+        assert Path("prime/foo.txt").is_file()
+
+        state_dir = Path("parts/foo/state")
+
+        assert sorted(state_dir.rglob("*")) == [
+            state_dir / "build",
+            state_dir / "prime",
+            state_dir / "pull",
+            state_dir / "stage",
+        ]
+
+        self._lifecycle.clean()
+
+        assert Path("parts").exists() is False
+        assert Path("stage").exists() is False
+        assert Path("prime").exists() is False
+
+        assert list(state_dir.rglob("*")) == []
+
+    def test_clean_part(self):
+        actions = self._lifecycle.plan(Step.PRIME)
+
+        with self._lifecycle.action_executor() as ctx:
+            ctx.execute(actions)
+
+        foo_state_dir = Path("parts/foo/state")
+        bar_state_dir = Path("parts/bar/state")
+
+        self._lifecycle.clean(part_names=["foo"])
+
+        assert Path("parts/foo/src/foo.txt").is_file() is False
+        assert Path("parts/foo/install/foo.txt").is_file() is False
+        assert Path("stage/foo.txt").is_file() is False
+        assert Path("prime/foo.txt").is_file() is False
```

Comment:
The test file could be inside a partition to reflect the real scenario.

Reply:
+1.  I will move this into a partition in a subsequent PR when more partition functionality is added.

(this entire integration test will grow with upcoming PRs)